src/types/presets.ts

```diff
@@ -45,5 +45,8 @@ export interface VercelBuildConfigV3 {
 export interface PresetOptions {
   vercel: {
     config: VercelBuildConfigV3;
+    function: {
+      runtime: string;
```

Comment:
Since we are only dealing with node, maybe simplify option to `nodeVersion`?

Reply:
Good point! My intention was because this setting essentially allows users to set different Node versions between runtime version at build time and serverless environment (although they should be the same), I wanted to make it clear that it is intended for serverless functions, if that makes sense...!

Reply:
`runtimeNodeVersion` should also be a good option 👍🏼 (i mainly like to avoid nesting config when not necessary)

Reply:
@smaeda-ks @pi0 I think this is a bad idea. The build time an runtime must match. So you shouldn't let the user configure it unless you can also swap out the build version of Node.js too.

Reply:
@styfle with this approach, you theoretically could:
- set the node version in vercel / engines: node in package.json (build)
- set vercel node version with new config param (runtime)
That would lead to different node versions by configuration. Not sure if anybody needs that but it would probably ship a faster "fix" to our problem right now, like @pi0 mentioned in his comment above.

Reply:
Is there any special reason for this matching @styfle other than natives matching? Nitro uses rollup/esbuild which can generates backward/forward compatible syntax based on expected runtime.

Reply:
@pi0 The reason this PR exists at all is because there was a mismatch between build and runtime. The reason mismatches cause problems native `dependencies` will break (basically any dependency that isn't pure JS).

Reply:
Then maybe we should rename the option to `vercel.nodeVersion` and make a runtime warning if build-time node is not matching?

Reply:
That sounds painful for the user. I suggest using `@vercel/build-utils` to set the correct PATH env before executing any `npm install` or `npm run build`. 

Here's an example:

https://github.com/vercel/vercel/blob/e8385566fad2ff56efec64fd7250909789662790/packages/next/src/index.ts#L258-L259

Reply:
Thanks! Going to remove this additon and use the `getNodeVersion` helper function instead👍